File: src/de/saschahlusiak/freebloks/controller/SpielClientInterface.java
package de.saschahlusiak.freebloks.controller;

import de.saschahlusiak.freebloks.model.Spiel;
import de.saschahlusiak.freebloks.model.Stone;
import de.saschahlusiak.freebloks.model.Turn;
import de.saschahlusiak.freebloks.network.NET_CHAT;
import de.saschahlusiak.freebloks.network.NET_SERVER_STATUS;
import de.saschahlusiak.freebloks.network.NET_SET_STONE;

public interface SpielClientInterface {
	void onConnected(Spiel spiel);
	void onDisconnected(Spiel spiel);

	void newCurrentPlayer(int player);
	void stoneWillBeSet(NET_SET_STONE s);
	void stoneHasBeenSet(NET_SET_STONE s);
	void hintReceived(NET_SET_STONE s);
	void gameFinished();
	void chatReceived(NET_CHAT c);
	void gameStarted();
	void stoneUndone(Stone s, Turn t);
	void serverStatus(NET_SERVER_STATUS status);
}

File: src/main/java/de/saschahlusiak/freebloks/controller/SpielClientInterface.kt
package de.saschahlusiak.freebloks.controller

import de.saschahlusiak.freebloks.model.Spiel
import de.saschahlusiak.freebloks.model.Turn
import de.saschahlusiak.freebloks.network.message.MessageServerStatus

interface SpielClientInterface {
    fun onConnected(spiel: Spiel)
    fun onDisconnected(spiel: Spiel)
    fun newCurrentPlayer(player: Int)
    fun stoneWillBeSet(turn: Turn)
    fun stoneHasBeenSet(turn: Turn)
    fun hintReceived(turn: Turn)
    fun gameFinished()
    fun chatReceived(client: Int, message: String)
    fun gameStarted()
    fun stoneUndone(t: Turn)
    fun serverStatus(status: MessageServerStatus)
}
